projects/admin/src/app/pipe/id-attribute.pipe.ts

```diff
@@ -0,0 +1,31 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'idAttribute'
+})
+
+export class IdAttributePipe implements PipeTransform {
+
+  transform(value: any, options?: {prefix?: string|null, suffix?: string|null}, ...args: any[]): any {
+    // If no options, return only value
+    if (!options) {
+      return value;
+    }
+
+    const result = [];
+    // Add prefix first (if exist)
+    if ('prefix' in options) {
+      result.push(options.prefix);
+    }
+
+    // Well, the initial value could be useful!
+    result.push(value);
+
+    // Add suffix at the end
+    if ('suffix' in options) {
+      result.push(options.suffix);
+    }
+    return result.join('-');
```

Comment:
You should write this compressed code :
```
let parts = [options.prefix, value, options.suffix];
parts = parts.filter(value=> value !== null);
return parts.join('-');
```
or even more compacted (but less human-readable):
```
return [options.prefix, value, options.suffix].filter(value=> value !== null).join('-');
```